app/experimenter/experiments/tests/test_forms.py

```diff
@@ -993,21 +993,21 @@ def test_form_is_invalid_if_branches_have_duplicate_pref_values(self):
         self.assertNotIn("value", form.variants_formset.errors[2])
 
     def test_form_is_invalid_if_pref_value_do_not_match_pref_type(self):
-        self.data["variants-0-value"] = "42"  # int
+        self.data["pref_type"] = Experiment.PREF_TYPE_INT
+        self.data["variants-0-value"] = "hello"  # str
         self.data["variants-1-value"] = "true"  # bool
-
+        self.data["variants-2-value"] = "5"  # int
         form = ExperimentVariantsPrefForm(request=self.request, data=self.data)
         self.assertFalse(form.is_valid())
         self.assertIn("value", form.variants_formset.errors[0])
         self.assertIn("value", form.variants_formset.errors[1])
         self.assertNotIn("value", form.variants_formset.errors[2])
 
-    def test_form_is_invalid_if_pref_value_has_invalid_json(self):
-        self.data["variants-0-value"] = "abc"  # missing quotes
-
+    def test_form_is_valid_if_pref_type_is_string(self):
```

Comment:
Can you add two more similar tests, one for int type and one for bool type just so we have our bases covered?

Reply:
will do